### PR TITLE
[bluetooth.bluez] Fill in manufacturer id when available

### LINKED ISSUE #4658: [bluetooth.bluez] BluetoothDevice manufacturerId always null
I'm building [Ruuvi Tag](https://ruuvi.com/) binding on top of the bluetooth addon. Since Ruuvi is advertising the sensor data, there seems to be many commonalities with `blukii` binding.

However, when testing my first PoC, I noticed that manufacturer ID seems to be always missing with bluez. This makes it hard to detect the Ruuvi sensors since the name of the device is just the address which is not suitable for detecting the sensor from rest of the devices. 

In `blukii` addon, the name is used to detect the sensor ([link](https://github.com/openhab/openhab2-addons/blob/master/addons/binding/org.openhab.binding.bluetooth.blukii/src/main/java/org/openhab/binding/bluetooth/blukii/internal/BlukiiDiscoveryParticipant.java#L51)) -- making it a non-issue there.

## Expected Behavior
When using `bluetooth.bluez`: `BluetoothDiscoveryParticipant` would be called with `BluetoothDevice` that has non-null `manufacturerId`.

## Current Behavior

`BluetoothDevice.getManufacturerId()` returns `null`.

Log excerpt below from a bundle that logs all `BluetoothDevice` information. You can see that Ruuvi tag, e.g. `F5:A6:5F:B9:2C:CC`, is sending manufacturer data, identified by manufacturer id `99 04`, 

However, when `BluetoothDiscoveryParticipant` is called, the `getManufacturerId()` always returns `null`.

```
05:35:52.540 [DEBUG] [.bluetooth.bluez.BlueZBluetoothDevice] - Received manufacturer data for 'F5:A6:5F:B9:2C:CC': 99 04 03 28 16 13 C5 F2 FF E6 FF C5 03 F6 0B B9
05:35:52.664 [DEBUG] [.bluetooth.bluez.BlueZBluetoothDevice] - Received manufacturer data for 'EC:76:F6:EF:DE:7B': 99 04 03 5B 05 4F C6 3F FF F6 FF F1 04 13 0B 7D
05:35:53.204 [DEBUG] [.bluetooth.bluez.BlueZBluetoothDevice] - Received manufacturer data for 'C9:07:79:14:78:15': 99 04 03 9F 86 10 C6 E1 00 01 FF E4 03 FB 0B 53
05:35:54.844 [DEBUG] [ooth.bluez.handler.BlueZBridgeHandler] - Refreshing Bluetooth device list...
05:35:55.032 [DEBUG] [ooth.bluez.handler.BlueZBridgeHandler] - Found 30 Bluetooth devices.
05:35:55.037 [INFO ] [internal.RuuviTagDiscoveryParticipant] - manufacturer id: null, name: 6C-D3-A3-21-EB-FE, address: 6C:D3:A3:21:EB:FE, services: []
05:35:55.046 [INFO ] [internal.RuuviTagDiscoveryParticipant] - manufacturer id: null, name: 57-D5-08-2A-04-F9, address: 57:D5:08:2A:04:F9, services: []
05:35:55.061 [INFO ] [internal.RuuviTagDiscoveryParticipant] - manufacturer id: null, name: 75-12-C7-99-69-63, address: 75:12:C7:99:69:63, services: []
05:35:55.076 [INFO ] [internal.RuuviTagDiscoveryParticipant] - manufacturer id: null, name: [Samsung] Soundbar J-Series, address: FC:F1:36:15:BA:8C, services: []
05:35:55.091 [INFO ] [internal.RuuviTagDiscoveryParticipant] - manufacturer id: null, name: 2C-B4-3A-16-CD-59, address: 2C:B4:3A:16:CD:59, services: []
05:35:55.104 [INFO ] [internal.RuuviTagDiscoveryParticipant] - manufacturer id: null, name: 4E-F0-C6-39-5C-E9, address: 4E:F0:C6:39:5C:E9, services: []
05:35:55.119 [INFO ] [internal.RuuviTagDiscoveryParticipant] - manufacturer id: null, name: EC-76-F6-EF-DE-7B, address: EC:76:F6:EF:DE:7B, services: []
05:35:55.132 [INFO ] [internal.RuuviTagDiscoveryParticipant] - manufacturer id: null, name: C9-07-79-14-78-15, address: C9:07:79:14:78:15, services: []
05:35:55.146 [INFO ] [internal.RuuviTagDiscoveryParticipant] - manufacturer id: null, name: 7B-28-32-26-63-D4, address: 7B:28:32:26:63:D4, services: []
05:35:55.159 [INFO ] [internal.RuuviTagDiscoveryParticipant] - manufacturer id: null, name: [TV]Samsung Matteux, address: 90:F1:AA:1F:91:61, services: []
05:35:55.172 [INFO ] [internal.RuuviTagDiscoveryParticipant] - manufacturer id: null, name: PHL-65PUS8303/12, address: C0:D2:F3:A7:F9:F2, services: []
05:35:55.185 [INFO ] [internal.RuuviTagDiscoveryParticipant] - manufacturer id: null, name: 79-4C-9C-5E-F2-D4, address: 79:4C:9C:5E:F2:D4, services: []
05:35:55.200 [INFO ] [internal.RuuviTagDiscoveryParticipant] - manufacturer id: null, name: HTC BS C5B21F, address: 40:4E:36:DD:05:C5, services: []
05:35:55.214 [INFO ] [internal.RuuviTagDiscoveryParticipant] - manufacturer id: null, name: 60-33-A2-1E-58-50, address: 60:33:A2:1E:58:50, services: []
05:35:55.227 [INFO ] [internal.RuuviTagDiscoveryParticipant] - manufacturer id: null, name: 42-F5-D2-01-BF-0C, address: 42:F5:D2:01:BF:0C, services: []
05:35:55.241 [INFO ] [internal.RuuviTagDiscoveryParticipant] - manufacturer id: null, name: 69-A9-8A-7E-3F-B6, address: 69:A9:8A:7E:3F:B6, services: []
05:35:55.255 [INFO ] [internal.RuuviTagDiscoveryParticipant] - manufacturer id: null, name: 4B-2A-4A-03-A7-A2, address: 4B:2A:4A:03:A7:A2, services: []
05:35:56.610 [INFO ] [internal.RuuviTagDiscoveryParticipant] - manufacturer id: null, name: Forerunner 235, address: FE:92:29:0F:E2:6A, services: []
05:35:56.624 [INFO ] [internal.RuuviTagDiscoveryParticipant] - manufacturer id: null, name: 5D-60-D8-9E-21-63, address: 5D:60:D8:9E:21:63, services: []
05:35:56.687 [INFO ] [internal.RuuviTagDiscoveryParticipant] - manufacturer id: null, name: 70-DA-ED-0D-E9-08, address: 70:DA:ED:0D:E9:08, services: []
05:35:56.707 [INFO ] [internal.RuuviTagDiscoveryParticipant] - manufacturer id: null, name: 5C-FA-7F-9E-81-26, address: 5C:FA:7F:9E:81:26, services: []
05:35:56.720 [INFO ] [internal.RuuviTagDiscoveryParticipant] - manufacturer id: null, name: F5-A6-5F-B9-2C-CC, address: F5:A6:5F:B9:2C:CC, services: []
05:35:56.733 [INFO ] [internal.RuuviTagDiscoveryParticipant] - manufacturer id: null, name: 52-E5-DE-C8-89-48, address: 52:E5:DE:C8:89:48, services: []
05:35:56.747 [INFO ] [internal.RuuviTagDiscoveryParticipant] - manufacturer id: null, name: 98-D6-BB-25-0B-F9, address: 98:D6:BB:25:0B:F9, services: []
05:35:56.760 [INFO ] [internal.RuuviTagDiscoveryParticipant] - manufacturer id: null, name: 9C-20-7B-93-3B-9A, address: 9C:20:7B:93:3B:9A, services: []
05:35:56.773 [INFO ] [internal.RuuviTagDiscoveryParticipant] - manufacturer id: null, name: HTC BS 8659BB, address: 40:4E:36:DC:E3:86, services: []
05:35:56.786 [INFO ] [internal.RuuviTagDiscoveryParticipant] - manufacturer id: null, name: 74-51-7C-D4-C4-A4, address: 74:51:7C:D4:C4:A4, services: []
05:35:56.799 [INFO ] [internal.RuuviTagDiscoveryParticipant] - manufacturer id: null, name: 6B-F7-D4-54-F5-6B, address: 6B:F7:D4:54:F5:6B, services: []
05:35:56.812 [INFO ] [internal.RuuviTagDiscoveryParticipant] - manufacturer id: null, name: 55-4D-68-2A-71-BF, address: 55:4D:68:2A:71:BF, services: []
05:35:56.825 [INFO ] [internal.RuuviTagDiscoveryParticipant] - manufacturer id: null, name: C8-69-CD-C7-F5-6A, address: C8:69:CD:C7:F5:6A, services: []
```

## Possible Solution

I think I can workaround the issue by introducing "intermediate" thing, "Ruuvi Tag Listener" which would subscribe to beacon messages, similar to [`BeaconBluetoothHandler` in `initialize`](https://github.com/openhab/openhab2-addons/blob/master/addons/binding/org.openhab.binding.bluetooth/src/main/java/org/openhab/binding/bluetooth/BeaconBluetoothHandler.java#L84). This "Ruuvi Tag Listener" would then discover all the Ruuvi tags and introduce them to the thing inbox.

However, I feel this add quite much complexity (one additional thing), and it should be possible to detect the things right away if the manufacturer Id would be present.

Or is there some other discovery "hook" that I am perhaps missing? Can I for example "participate in the discovery" by listening beacon messages right away?

## Steps to Reproduce (for Bugs)

1. add log statement to [`BlukiiDiscoveryParticipant.getThingUID`](https://github.com/openhab/openhab2-addons/blob/master/addons/binding/org.openhab.binding.bluetooth.blukii/src/main/java/org/openhab/binding/bluetooth/blukii/internal/BlukiiDiscoveryParticipant.java#L49) log bluetooth device manufacturer Id
2.

## Your Environment

* Version used: (e.g., openhab and addon versions)
* Environment name and version (e.g. Chrome 39, node.js 5.4, Java 8, ...):
* Operating System and version (desktop or mobile, Windows 10, Raspbian Jessie, ...):
Raspberry Pi 3 B, raspbian, openHAB 2.4.0 bluez release

### PR DESCRIPTION
**Pull request by ssalonen**
Resolves #4658

Signed-off-by: Sami Salonen <ssalonen@gmail.com>

### CODE REVIEW

**Comment by martinvw** [comment]
@cdjackson does this look okay to you? AFAIK you are the most sensible reviewer? But please let us know if we are wrong :-)

File: addons/binding/org.openhab.binding.bluetooth.bluez/src/main/java/org/openhab/binding/bluetooth/bluez/handler/BlueZBridgeHandler.java [hunk 1]
```diff
@@ -254,6 +254,15 @@ private void checkForNewDevices() {
 
     private BlueZBluetoothDevice createAndRegisterBlueZDevice(tinyb.BluetoothDevice tinybDevice) {
         BlueZBluetoothDevice device = new BlueZBluetoothDevice(this, tinybDevice);
+        tinybDevice.getManufacturerData().keySet().stream().filter(manufacturerId -> manufacturerId != null).findFirst()
+                .ifPresent(manufacturerId -> {
+                    if (manufacturerId != null) {
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.bluetooth.bluez/src/main/java/org/openhab/binding/bluetooth/bluez/handler/BlueZBridgeHandler.java]: It looks like this check should not be needed?

**Comment by cdjackson** [comment]
Yes, looks good to me in general.

File: addons/binding/org.openhab.binding.bluetooth.bluez/src/main/java/org/openhab/binding/bluetooth/bluez/handler/BlueZBridgeHandler.java (hunk 1 above)
Comment (ssalonen) [reply addons/binding/org.openhab.binding.bluetooth.bluez/src/main/java/org/openhab/binding/bluetooth/bluez/handler/BlueZBridgeHandler.java]: Oh right, nulls are filtered earlier. Will clean it up

Comment (ssalonen) [reply addons/binding/org.openhab.binding.bluetooth.bluez/src/main/java/org/openhab/binding/bluetooth/bluez/handler/BlueZBridgeHandler.java]: Now resolved in ca8d5e5

**Comment by martinvw** [comment]
> @matinvw It is up to you.

@cweitkamp might have missed that because of the typo ;-)

**Comment by cweitkamp** [comment]
@martinvw I am sorry for that. Typing on mobile phones sometimes corrects spelling incorrectly. :disappointed: And there is no preview for the user names.

**Comment by martinvw** [comment]
No problem, I was just disappointed in my self for missing it and then I saw that I had at least some excuse :-D

**Pull request merged**